scripts/common-functions.sh

```diff
@@ -74,6 +74,87 @@ function dht_key {
   $(find_binary cardano-dht-keygen) -n 000000000000$i2 | tr -d '\n'
 }
 
+# Generates kademliaN.yaml and topologyN.yaml for demo setup 
+# for N \in {0..n-1} where n is number of nodes specified.
+function gen_kademlia_topology {
+  local total_nodes=$1
+  local npred=$(($total_nodes-1))
+
+  if [[ $total_nodes -le 0 ]]; then
+    echo "gen_kademlia_topology: n should be positive, but is $total_nodes"
+    exit
+  fi
+
+  if [[ "$2" == "" ]]; then
+    local out_dir="./run/"
+  else
+    local out_dir=$2
+  fi
+
+  echo "Cleaning up topology/kademlia files"
+  rm -v $out_dir/topology*.yaml
+  rm -v $out_dir/kademlia*.yaml
+
+  echo "Generating new topology/kademlia files"
+  for i in $(seq 0 $total_nodes); do
+
+    # generate kademlia config
+
+    if [[ $i -eq $total_nodes ]]; then
+      local kfile=$out_dir/kademlia_explorer.yaml
+      local others_n=$total_nodes
+    else
+      local kfile=$out_dir/kademlia$i.yaml
+      local others_n=$npred
+    fi
+
+    if [[ $total_nodes -eq 1 ]]; then
+      echo "peers: []" > $kfile
+    else
+      echo "peers: " > $kfile
+    fi
+    
+    for j in $(seq 0 $others_n); do
+      if [[ $j -eq $i ]]; then continue; fi
+      echo "  - host: '127.0.0.1'" >> $kfile
+      echo "    port: 300$j"       >> $kfile
+    done
+
+    echo "address:" >> $kfile
+    echo "  host: '127.0.0.1'" >> $kfile
+    echo "  port: 300$i" >> $kfile
+
+
+    # generate topology config (it's the same for all nodes)
+
+    # we generate n-1 topology configs, there's no explorer topology (should there be?)
```

Comment:
I think we don't need _explorer_ in the topology since we are using it only as a relay node and not as a core node. I'm not sure though :man_shrugging:

Reply:
Well, the file was there (implying it is used by someone) so i came up with the generic way to generate it.

Reply:
Yes. And you did a good work. A similar thing occurred to me as well.
I was commenting on this - `there's no explorer topology (should there be?)`.

Reply:
Ah, sorry, i missed the point of your message. All topology files are the same so explorer can just use node's config (i guess). I also think that maybe I should generate one file instead of `n` files. I will implement this behaviour.

Reply:
Yes, seems to be so, good catch. The only reason why we would leave N nodes would be if we wanted to have separate configuration for each node.
I don't know why we would use that, other than test, but you can leave this behaviour as a separate function in case someone want's to use it like it works now and has separate configuration for each node.